lib/generators/solidus_stripe/install/templates/app/javascript/controllers/solidus_stripe_intent_controller.js

```diff
@@ -50,25 +50,20 @@ export default class extends Controller {
   }
 
   confirmIntent() {
+    let confirmMethod
+
+    if (this.flowValue === 'payment') confirmMethod = 'confirmPayment'
+    if (this.flowValue === 'setup') confirmMethod = 'confirmSetup'
```

Comment:
Do you know how it is going to play out with the strategy class?

Reply:
I tried to move the code around in a few ways, I thought we could actually pass the confirm function name from outside, but then I discarded it because it feels wrong to have the view know the JS call that should be used to confirm a payment. Eventually I stopped to passing the method name and normalizing the calls.